frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/NetworkForm.tsx

```diff
@@ -1,59 +1,85 @@
-    const [resourceJSON, setResourceJSON] = useState<any>()
-    const [editAgent, setEditAgent] = useState()
+const fields: any = {
+    apiVip: { label: 'API Virtual IP' },
+    ingressVip: { label: 'Ingress Virtual IP' },
+    clusterNetworkCidr: { label: 'Cluster network CIDR' },
+    clusterNetworkHostPrefix: { label: 'Cluster network host prefix' },
+    serviceNetworkCidr: { label: 'Service network CIDR' },
+    sshPublicKey: { label: 'Host SSH Public Key' },
+}
 
+const NetworkForm: React.FC<NetworkFormProps> = ({ control, handleChange }) => {
+    const formRef = useRef<FormikProps<NetworkConfigurationValues>>(null)
     useEffect(() => {
-        const resource = handleCreateResource(true) //await
-        setResourceJSON(resource)
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [])
-
-    const [clusterDeployments, agentClusterInstalls, agents] = useRecoilValue(
-        waitForAll([clusterDeploymentsState, agentClusterInstallsState, agentsState])
-    )
+        if (control.active) {
+            formRef?.current?.setValues(control.active, false)
+        } else {
+            control.active = formRef?.current?.values
+        }
+        control.validate = () => {
+            return formRef?.current?.submitForm().then(() => {
+                return formRef?.current?.errors
+            })
+        }
+        control.summary = () => {
+            return Object.keys(fields).map((key) => {
+                return {
+                    term: fields[key].label,
+                    desc: get(control, `active.${key}`),
+                    exception: get(control, `errors.${key}`),
+                }
+            })
+        }
+    }, [control])
 
-    const cdName = resourceJSON?.createResources.find((r: any) => r.kind === ClusterDeploymentKind).metadata.name
+    const [editAgent, setEditAgent] = useState()
+    const [agents] = useRecoilValue(waitForAll([agentsState]))
 
-    const clusterDeployment = clusterDeployments.find(
-        (c) => c.metadata.name === cdName && c.metadata.namespace === cdName
-    )
-    const agentClusterInstall = agentClusterInstalls.find(
-        (a) => a.metadata.name === cdName && a.metadata.namespace === cdName
-    )
+    const { resourceJSON = {} } = control
+    const { createResources = [] } = resourceJSON
+    const clusterDeployment = createResources.find((r: { kind: string }) => r.kind === 'ClusterDeployment')
+    const agentClusterInstall = createResources.find((r: { kind: string }) => r.kind === 'AgentClusterInstall')
 
     useEffect(() => (control.agentClusterInstall = agentClusterInstall), [control, agentClusterInstall])
 
     const onValuesChanged = useCallback((values) => {
-        control.active = values
-        handleChange(control)
-        // eslint-disable-next-line react-hooks/exhaustive-deps
+        if (!isEqual(values, control.active)) {
+            control.active = values
+            control.step.title.isComplete = false
```

Comment:
How is FormControl state managed? I'd expect the state to be set using setState or by dispatching actions depending on how the state is managed (useState or useReducer)